src/pages/ReimbursementAccount/ACHContractStep.js

```diff
@@ -114,57 +148,29 @@ class ACHContractStep extends React.Component {
             || (_.size(this.state.beneficialOwners) === 3 && !this.state.ownsMoreThan25Percent);
     }
 
-    /**
-     * Clear the error associated to inputKey if found and store the inputKey new value in the state.
-     *
-     * @param {Integer} ownerIndex
-     * @param {Object} values
-     */
-    clearErrorAndSetBeneficialOwnerValues(ownerIndex, values) {
-        this.setState((prevState) => {
-            const beneficialOwners = [...prevState.beneficialOwners];
-            beneficialOwners[ownerIndex] = {...beneficialOwners[ownerIndex], ...values};
-            BankAccounts.updateReimbursementAccountDraft({beneficialOwners});
-            return {beneficialOwners};
-        });
-
-        // Prepare inputKeys for clearing errors
-        const inputKeys = _.keys(values);
-
-        // dob field has multiple validations/errors, we are handling it temporarily like this.
-        if (_.contains(inputKeys, 'dob')) {
-            inputKeys.push('dobAge');
-        }
-        this.clearErrors(_.map(inputKeys, inputKey => `beneficialOwnersErrors.${ownerIndex}.${inputKey}`));
-    }
-
-    submit() {
-        if (!this.validate()) {
-            return;
-        }
-
+    submit(values) {
         const bankAccountID = lodashGet(store.getReimbursementAccountInSetup(), 'bankAccountID');
 
-        // If they did not select that there are other beneficial owners, then we need to clear out the array here. The
-        // reason we do it here is that if they filled out several beneficial owners, but then toggled the checkbox, we
-        // want the data to remain in the form so we don't lose the user input until they submit the form. This will
-        // prevent the data from being sent to the API
-        this.setState(prevState => ({
-            beneficialOwners: !prevState.hasOtherBeneficialOwners ? [] : prevState.beneficialOwners,
-        }),
-        () => BankAccounts.updateBeneficialOwnersForBankAccount({...this.state, beneficialOwners: JSON.stringify(this.state.beneficialOwners), bankAccountID}));
-    }
+        const beneficialOwners = !this.state.hasOtherBeneficialOwners ? []
+            : _.map(this.state.beneficialOwners, ownerID => ({
```

Comment:
Does this `map()` mainly serve the purpose of removing the "id" we have added?

Reply:
Yup, that's the idea. Similar to what we currently do [here](https://github.com/Expensify/App/blob/b2f768e65b88c17a6277f1538392874acf383d26/src/pages/ReimbursementAccount/ACHContractStep.js#L148-L154). Maybe we should add that comment back.

Reply:
yes, looks similar, but not in the way I am asking about (i.e. we are not removing any temporary id). Anyways, I am trying to think of some way to get rid of this id 😄 but I'll consolidate thoughts in the other thread when I understand the problem better.